packages/menu/src/useMenu.ts

```diff
@@ -39,25 +39,32 @@ import {
 } from './types';
 
 export const useMenu = <T extends HTMLElement = HTMLElement, M extends HTMLElement = HTMLElement>({
-  items: rawItems,
-  idPrefix,
+  defaultExpanded = false,
+  defaultFocusedValue,
+  document: documentProp,
   environment,
-  menuRef,
-  triggerRef,
-  rtl = false,
-  onChange = () => undefined,
+  focusedValue,
+  idPrefix,
   isExpanded,
-  defaultExpanded = false,
+  items: rawItems,
+  menuRef,
   restoreFocus = true,
+  rtl = false,
   selectedItems,
-  focusedValue,
-  defaultFocusedValue
+  triggerRef,
+  window: windowProp,
+  onChange = () => undefined
 }: IUseMenuProps<T, M>): IUseMenuReturnValue => {
   const prefix = `${useId(idPrefix)}-`;
   const triggerId = `${prefix}menu-trigger`;
   const isExpandedControlled = isExpanded !== undefined;
   const isSelectedItemsControlled = selectedItems !== undefined;
   const isFocusedValueControlled = focusedValue !== undefined;
+
+  const _window = windowProp || environment;
```

Comment:
I recall there being problems with initializing this outside of the handlers because it potentially references `globals` in a way that breaks SSR. The logic should be restored throughout to get the `window` and/or `document` as needed – probably via a utility function.